staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod_expansion.go

```diff
@@ -43,3 +45,18 @@ func (c *pods) Evict(eviction *policy.Eviction) error {
 func (c *pods) GetLogs(name string, opts *v1.PodLogOptions) *restclient.Request {
 	return c.client.Get().Namespace(c.ns).Name(name).Resource("pods").SubResource("log").VersionedParams(opts, scheme.ParameterCodec)
 }
+
+// PatchStatus modifies the status of an existing pod. It returns the copy of
+// the pod that the server returns, or an error.
+func (c *pods) PatchStatus(name string, data []byte) (*v1.Pod, error) {
+	result := &v1.Pod{}
+	err := c.client.Patch(types.StrategicMergePatchType).
```

Comment:
pass the patch type in as a parameter

Reply:
why can't just `Patch` be used for this? It passes the patch type and `subresources` as a parameter.

https://github.com/kubernetes/kubernetes/blob/aa61d4f080a2be9dedf9a17e3303e6ede937fb7b/staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod.go#L163

Reply:
good point

Reply:
that's not a great pattern to emulate (doesn't include patch type, requires a new method every time you wanted to patch a new subresource)

Reply:
Can we deprecate `node.PatchStatus`?

Reply:
Okay. What if I just write a bunch helper functions for patching status under `/kubernetes/pkg/util/pod/pod.go`